### PR TITLE
Travis changes

### PR DESCRIPTION
**Pull request by cltnschlosser**
Separates build using travis stages, adds independent javadoc check, and moves dropbox upload into only on build success.

Most of this pull request in javadoc fixes.

### CODE REVIEW

File: src/main/java/com/minecolonies/entity/EntityFishHook.java [hunk 1]
```diff
@@ -736,6 +736,7 @@ private void spawnLootAndExp(final EntityCitizen citizen)
      * @param citizen the fisherman getting the loot
      * @return an ItemStack randomly from the loot table
      */
+    @SuppressWarnings("unchecked")
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/entity/EntityFishHook.java]: INFO Suppressing warnings is not allowed [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1309)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [BlockParticleEffectMessage.java](https://github.com/Minecolonies/minecolonies/blob/bf9f224eef014f969a5d8787a40dbc61cd0c160c/src/main/java/com/minecolonies/network/messages/BlockParticleEffectMessage.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

**Comment by Kostronor** [comment]
LGTM just wanted to understand the changes @cltnschlosser

**Comment by cltnschlosser** [comment]
:shipit:

**Comment by Kostronor** [comment]
@Raycoms can we merge this?

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [BlockParticleEffectMessage.java](https://github.com/Minecolonies/minecolonies/blob/48837cdd753238b4aa32ed4ce23d470b890ff6e9/src/main/java/com/minecolonies/network/messages/BlockParticleEffectMessage.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [BlockParticleEffectMessage.java](https://github.com/Minecolonies/minecolonies/blob/8ea0ee061123f49fda895a8febeb3d813f353a36/src/main/java/com/minecolonies/network/messages/BlockParticleEffectMessage.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/44/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [BlockParticleEffectMessage.java](https://github.com/Minecolonies/minecolonies/blob/af84c109ab10935a037b42879d35fb81d35e97e9/src/main/java/com/minecolonies/network/messages/BlockParticleEffectMessage.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [BlockParticleEffectMessage.java](https://github.com/Minecolonies/minecolonies/blob/405db5b4ab0ba629965c223a4baaec5b58de87e2/src/main/java/com/minecolonies/network/messages/BlockParticleEffectMessage.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [BlockParticleEffectMessage.java](https://github.com/Minecolonies/minecolonies/blob/6aa6495c43b05f3e5e7c1bc1ed1ef4818c50faad/src/main/java/com/minecolonies/network/messages/BlockParticleEffectMessage.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [BlockParticleEffectMessage.java](https://github.com/Minecolonies/minecolonies/blob/f864dac9f7f54add3384d115dd8f9c650aef218a/src/main/java/com/minecolonies/network/messages/BlockParticleEffectMessage.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [BlockParticleEffectMessage.java](https://github.com/Minecolonies/minecolonies/blob/2751d1a403734830ad405e534360d8570da242b8/src/main/java/com/minecolonies/network/messages/BlockParticleEffectMessage.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [BlockParticleEffectMessage.java](https://github.com/Minecolonies/minecolonies/blob/2a6b387673fb48ef1ad885b61abcf87f646ba153/src/main/java/com/minecolonies/network/messages/BlockParticleEffectMessage.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [BlockParticleEffectMessage.java](https://github.com/Minecolonies/minecolonies/blob/59a7e8f5947aaad35f9ad63e246697c1072623c9/src/main/java/com/minecolonies/network/messages/BlockParticleEffectMessage.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [BlockParticleEffectMessage.java](https://github.com/Minecolonies/minecolonies/blob/6fa6a113df9b33e82458db910288471c72b24482/src/main/java/com/minecolonies/network/messages/BlockParticleEffectMessage.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [BlockParticleEffectMessage.java](https://github.com/Minecolonies/minecolonies/blob/ea36941cce4b51e3b48e5c718f9286f6b46abf6c/src/main/java/com/minecolonies/network/messages/BlockParticleEffectMessage.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

File: .travis.yml [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
-install: true
 jdk: oraclejdk8
-script: ./travis.sh
+install: ./gradlew build
+script:
+  - ./gradlew javadoc
+  - ./sonar.sh
+after_success: ./dropbox.sh
 before_cache:
-- rm -rf $HOME/.gradle/caches/modules-2/modules-2.lock
```
Comment (cltnschlosser) [inline .travis.yml]: Should we just delete the whole .gradle folders? Also why do -r for files?

File: sonar.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+# filter out # for sonar
+BRANCH=$(echo $TRAVIS_BRANCH | sed 's/[#]//g')
```
Comment (cltnschlosser) [inline sonar.sh]: I'd prefer ```| tr -d '#'```, because is sed is kinda gross, but its aright

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [BlockParticleEffectMessage.java](https://github.com/Minecolonies/minecolonies/blob/2a3eb81c0d308d8b558209b24966ff2f93990a58/src/main/java/com/minecolonies/network/messages/BlockParticleEffectMessage.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

File: dropbox.sh [hunk 4]
```diff
@@ -1,4 +1,6 @@
-rm -f ~/.dropbox_uploader
+#!/bin/bash
+
+rm -rfv ~/.dropbox_uploader
```
Comment (cltnschlosser) [inline dropbox.sh]: again why -r for a single file.

**Comment by Kostronor** [comment]
:shipit: Fixed up all the cache things and now builds are way faster! @cltnschlosser @Raycoms

File: .travis.yml (hunk 2 above)
Comment (Kostronor) [inline .travis.yml]: No, we want the folder cached and only delete the stuff that changes every build

File: dropbox.sh (hunk 4 above)
Comment (Kostronor) [inline dropbox.sh]: because it does no harm and was a way to make travis rebuild the thing ;) If it annoys you I can change them all back

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 2 issues
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [BlockParticleEffectMessage.java](https://github.com/Minecolonies/minecolonies/blob/e1b941b40c8d6e7354e5cac130020e70bb771f1d/src/main/java/com/minecolonies/network/messages/BlockParticleEffectMessage.java): Add a new line at the end of this file. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS00113)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 1 issue
* INFO 1 info

Watch the comments in this conversation to review them.

File: src/main/java/com/minecolonies/entity/EntityFishHook.java [hunk 5]
```diff
@@ -736,7 +736,6 @@ private void spawnLootAndExp(final EntityCitizen citizen)
      * @param citizen the fisherman getting the loot
      * @return an ItemStack randomly from the loot table
      */
-    @SuppressWarnings("unchecked")
     private ItemStack getFishingLoot(final EntityCitizen citizen)
```
Comment (cltnschlosser) [inline src/main/java/com/minecolonies/entity/EntityFishHook.java]: We should probably actually check our casts then, so we don't get all the warnings.

Comment (Kostronor) [inline src/main/java/com/minecolonies/entity/EntityFishHook.java]: I removed the suppression so we we will get warnings once we change lines there (or look into sonar)

**Comment by Raycoms** [comment]
:+1:

**Pull request merged**